spec/support/shared_examples/structured_data_2_text_preprocessor.rb

```diff
@@ -0,0 +1,764 @@
+          </td>
+          </tr>
+          </table>
+          </div></div></td>
+          <td class="tableblock halign-left valign-top"><div class="content"><div class="paragraph">
+          <p>Academica Sinica&#8201;&#8212;&#8201;Chinese Tongyong Pinyin (2002)</p>
+          </div></div></td>
+          </tr>
+          </tbody>
+          </table>
+          </div>
+          </div>
+        TEXT
+      end
+
+      it "correctly renders input" do
+        expect(Asciidoctor.convert(input) + "\n").to(eq(output))
+      end
+    end
+
+    context "Liquid code snippets" do
+      let(:example_content) do
+        [{ "name" => "One", "show" => true },
+         { "name" => "Two", "show" => true },
+         { "name" => "Three", "show" => false }]
+      end
+      let(:input) do
+        <<~TEXT
+          = Document title
+          Author
+          :docfile: test.adoc
+          :nodoc:
+          :novalid:
+          :no-isobib:
+          :imagesdir: spec/assets
+
+          == base
+
+          [#{extention}2text,#{example_file},my_context]
+          ----
+          {% for item in my_context %}
+          {% if item.show %}
+          {{ item.name | upcase }}
+          {{ item.name | size }}
+          {% endif %}
+          {% endfor %}
+          ----
+        TEXT
+      end
+      let(:output) do
+        <<~TEXT
+          <div class='sect1'>
+            <h2 id='_base'>base</h2>
+            <div class='sectionbody'>
+              <div class='paragraph'>
+                <p>ONE 3</p>
+              </div>
+              <div class='paragraph'>
+                <p>TWO 3</p>
+              </div>
+            </div>
+          </div>
+        TEXT
+      end
+
+      it "renders liquid markup" do
+        expect(xmlpp(Asciidoctor.convert(input))).to(eq(xmlpp(output)))
+      end
+    end
+
+    context "Date time objects support" do
+      let(:example_content) do
+        { "date" => Date.parse("1889-09-28"), "time" => Time.gm(2020, 10, 15, 5, 34) }
+      end
+      let(:input) do
+        <<~TEXT
+          = Document title
+          Author
+          :docfile: test.adoc
+          :nodoc:
+          :novalid:
+          :no-isobib:
+          :imagesdir: spec/assets
+
+          == base
+
+          [#{extention}2text,#{example_file},my_context]
+          ----
+          {{my_context.time}}
+
+          {{my_context.date}}
+          ----
+        TEXT
+      end
+      let(:output) do
+        <<~TEXT
+          <div class='sect1'>
+            <h2 id='_base'>base</h2>
+            <div class='sectionbody'>
+              <div class='paragraph'>
+                <p>2020-10-15 05:34:00 UTC</p>
+              </div>
+              <div class='paragraph'>
+                <p>1889-09-28</p>
+              </div>
+            </div>
+          </div>
+        TEXT
+      end
+
+      it "renders date time objects" do
+        expect(xmlpp(Asciidoctor.convert(input))).to(eq(xmlpp(output)))
```

Comment:
This should be calling Metanorma instead of Asciidoctor. We do not want to write an Asciidoctor plugin.

Reply:
Rwrtied to use `Metanorma::Input::Asciidoc` instead